activerecord/lib/active_record/persistence.rb

```diff
@@ -555,6 +555,10 @@ def destroy_row
     end
 
     def relation_for_destroy
+      relation_for_itself
+    end
+
+    def relation_for_itself
```

Comment:
We should try to avoid adding methods to `ActiveRecord::Base` since they could conflict with user methods. Can we put an underscore in front of this?